### PR TITLE
Updates for terraform version 12; updated and validated

### PR DESCRIPTION
**Pull request by scottpgallagher**

### CODE REVIEW

**Comment by bdausses** [comment]
Full disclosure, I've not spun this up yet...  Am at a customer site all week, but can we clean this up?  

``` 
azure/main.tf
2:  required_version = "~> 0.11"

azure/versions.tf
3:  required_version = ">= 0.12"

chef-automate/azure/versions.tf
3:  required_version = ">= 0.12"

chef-automate/azure/resource_group.tf
2:  required_version = "~> 0.11"

chef-automate/aws/data.tf
2:#   required_version = "~> 0.11.11" // Terraform frequently puts breaking changes into minor and patch version releases. _Always_ hard pin to the latest known and tested working version. Do not trust semantic versioning.

chef-automate/aws/versions.tf
3:  required_version = ">= 0.12"

gke/versions.tf
3:  required_version = ">= 0.12"

aws/versions.tf
3:  required_version = ">= 0.12"
```

Also, I super highly uberly suggest we add `.tfswitch` files in here.  What's tfswitch?  --> . https://chefio.slack.com/archives/CCBU6CFF0/p1569618883218800

Actually, the tfswitch comment applies to ANY terraform code we're writing.  The tfswitch tool is so easy to use and adding this file automatically makes sure you're using the right version for a given plan.  If you're not using tfswitch, this file doesn't do anything, so no reason to not have it.

**Comment by scottpgallagher** [comment]
@bdausses did those changes. I didn't add the .tfswitch stuff though to this commit.

**Comment by bdausses** [comment]
@scottpgallagher Per our Zoom call:

- Lets add back version pinnings, but make sure they're `>= 0.12`
- Update the `tfvars.example` files to reflect `node_count` rather than `count`
- Update the Azure plan to make `azureuser` be able to do passwordless `sudo` commands. (I already have the code for this)

I'll put something on the calendar for next week for us.

**Comment by scottpgallagher** [comment]
We will also need to see where else node is referenced and update those items to node_count, I remember there being a tf file that has those in it.

**Comment by scottpgallagher** [comment]
Added in .tfswitchrc files

**Comment by ChefRycar** [comment]
I've fixed merge conflicts with master now that the comments PR has been merged. 
I've also removed the `hab install chef/applications-service` step from the chef-automate plan, since it's now part of the regular install process. This removes a now-extraneous `sleep 60`, which should speed things up, and means we don't need to do a license accept on the automate TF.

**Comment by ericcalabretta** [comment]
@ChefRycar Is just the linting in visual studio the concern or does that have another impact? As long as the code actually works I see the benefits of being on current way higher than the full TF ecosystem being ready. 

The `hab install` command is breaking the chef-automate plan now too. Automate upgraded to a post license agreement version of Habitat, so we either need to remove the `hab install` command or add a license accept prior to it.

**Comment by ChefRycar** [comment]
Yup, just the code linting. There's some experimental support, but it's pretty experimental. 
tl;dr: Everything works fine for me on latest 12. Only oddity will be some false positive warnings if you're using the TF plugin in VSC for a bit.
